### DISCUSSION TITLE
Welcome to oci-native-ingress-controller Discussions!

### DISCUSSION THREAD

**Discussion by nirpai** [oracle/oci-native-ingress-controller#70] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by khchoy** [comment]
Hi,
Do you guys have any plan to merge this PR. I need this feature urgently. Thanks.
[Feature] Add extract-ca-from-tls-crt annotation (#60) #61

**Reply by nirpai** [reply]
I will prioritize this for a upcoming release.

**Reply by khchoy** [reply]
> I will prioritize this for a upcoming release.

Is it possible to extend this feature to Load Balancer type Service's secret? For example, this annotation used by Service, i.e., "service.beta.kubernetes.io/oci-load-balancer-tls-backendset-secret"

**Comment by khchoy** [comment]
Hi I tested v1.3.7, still not up to my expectations.
Firstly, I pre-created a load balancer with a http listener (port 80) via terraform script and I assign the LB ocid to the IngressClass's annotation. IngressClass was installed. Then I deploy the Ingress resources with service port 443. My expectation is that I should have two listeners at the end of the deployment. Finally, everything created normally in the LB except the http listener (port 80) get deleted and ended up with only https listener (port 443) available. Why the http listener got deleted after Ingress resource deployment ended? Secondly, I took 2nd attempt and tried with these new annotation "oci-native-ingress.oraclecloud.com/http-listener-port: "80"" and "oci-native-ingress.oraclecloud.com/https-listener-port: "443"". I am expecting to have two listeners created at the end of the Ingress resource deployment. However, there is only one listener (port 443) get created. Why listener (port 80) not created in LB?

**Reply by nirpai** [reply]
Please share your ingress definition.

**Reply by khchoy** [reply]
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: desktop-online
  labels:
    chart: usage-engine-private-edition-4.2.0-PR-3179-20240529022911-c2d58
    release: uepe420pr3179
  annotations:
    oci-native-ingress.oraclecloud.com/healthcheck-protocol: HTTP
    oci-native-ingress.oraclecloud.com/healthcheck-return-code: "200"
    oci-native-ingress.oraclecloud.com/http-listener-port: "80"
    oci-native-ingress.oraclecloud.com/https-listener-port: "443"
    oci-native-ingress.oraclecloud.com/policy: ROUND_ROBIN
    oci-native-ingress.oraclecloud.com/protocol: HTTP
    oci-native-ingress.oraclecloud.com/certificate-ocid: ocid1.certificate.oc1.eu-frankfurt-1.amaaaaaaqpnxi2aaftofigjmkytoomv2u2ycjenhvqsbarhfhpycfujihyyq
    oci-native-ingress.oraclecloud.com/healthcheck-path: "/desktop/"
    oci-native-ingress.oraclecloud.com/healthcheck-port: "9001"
spec:
  ingressClassName: native-ic-ingress-class
  rules:
  - host: "desktop-online.test-uepe-cluster-1.stratus.oci.digitalroute.net"
    http:
      paths:
      - pathType: Prefix
        path: "/"
        backend:
          service:
            name: desktop-online
            port:
              number: 443

**Reply by nirpai** [reply]
You are using `oci-native-ingress.oraclecloud.com/certificate-ocid `annotation. This would make all paths defined in your ingress as https. If you need http functionality move the path to a new ingress without the annotation declation. If you need both http and https in same ingress you need to configure the tls section and use secrets. Then only paths mapped to the tls declared hosts would be https enabled.

**Reply by khchoy** [reply]
The reason I need http (port 80) listener exists is to configure https redirection with LB Rule set. Is there any annotation for https redirection ?

**Reply by nirpai** [reply]
No we don't support redirections currently. Willl keep you posted if we add the redirection support.

**Reply by khchoy** [reply]
I can't use Ingress resource with secret right now as it has limitation mentioned in this PR, https://github.com/oracle/oci-native-ingress-controller/pull/61. Wondering if this PR will be included in next release, i.e., v1.3.8?

**Reply by nirpai** [reply]
Hopefully yes. Will keep you posted.

**Reply by khchoy** [reply]
Hi, I saw this PR, https://github.com/oracle/oci-native-ingress-controller/pull/61 already been merged to main branch last two days. Can I test with latest main branch now or I still need to wait for official release?

**Reply by nirpai** [reply]
This will be part of our next release. We don't have an ETA for the release yet.

**Reply by nirpai** [reply]
Available now as part of 1.3.8

**Reply by pr07pr07** [reply]
Hi @nirpai ,

There are some vulnerabilities in the Ingress controller. Do we need to take any action from our side when using the OCI-native Ingress controller?

Refer: https://thehackernews.com/2025/03/critical-ingress-nginx-controller.html

Currently using this version :
oci-native-ingress-controller:v1.3.3

Your guidance on this would be greatly appreciated.
Thanks in advance!

Kr,
Prathap

**Reply by nirpai** [reply]
This is a vulnerability reported in nginx ingress controller. For NIC, it is always recommended to upgrade to latest version of software.

**Reply by pr07pr07** [reply]
Thank you for your prompt response, @nirpai 
We will plan to upgrade it as soon as possible as per the recommendation.
